iset.mm

```diff
@@ -34966,6 +34966,126 @@ than an implication (but the two are equivalent by ~ bm1.3ii ).
+       we instead work with ~ ax-bdsep , ` EXMID ` as defined here means that
+       all bounded propositions are decidable.
+
+       (Contributed by Mario Carneiro and Jim Kingdon, 18-Jun-2022.) $)
+    df-exmid $a |- ( EXMID <-> A. x ( x C_ { (/) } -> DECID (/) e. x ) ) $.
+  $}
+
+  ${
+    $d ph x z $.
+    $( EXMID implies that an arbitrary proposition is decidable.  That is,
+       EXMID captures the usual meaning of excluded middle when stated in terms
+       of propositions.
+
+       To get other propositional statements which are equivalent to excluded
+       middle, combine this with ~ notnotrdc , ~ peircedc , or ~ condc .
+
+       (Contributed by Jim Kingdon, 18-Jun-2022.) $)
+    exmidexmid $p |- ( EXMID -> DECID ph ) $=
+      ( vz vx wem c0 csn crab wcel wdc wss ssrab2 cv wi wal df-exmid p0ex rabex
+      wceq sseq1 eleq2 dcbid imbi12d spcv sylbi mpi 0ex biidd elrab dcbii sylib
+      snid mpbiran ) DEABEFZGZHZIZAIDUNUMJZUPABUMKDCLZUMJZEURHZIZMZCNUQUPMZCOVB
+      VCCUNABUMPQURUNRZUSUQVAUPURUNUMSVDUTUOURUNETUAUBUCUDUEUOAUOEUMHAEUFUKAABE
+      UMBLERAUGUHULUIUJ $.
+  $}
+
+  ${
+    $d x y z $.
+    $( Excluded middle is equivalent to saying any subset of ` { (/) } ` is
+       either ` (/) ` or ` { (/) } ` .  (Contributed by BJ and Jim Kingdon,
+       18-Jun-2022.) $)
+    exmid01 $p |- ( EXMID <-> A. x ( x C_ { (/) }
+        -> ( x = (/) \/ x = { (/) } ) ) ) $=
+      ( vy vz wem cv c0 csn wss wcel wdc wi wal wo df-exmid wn wa eleq2 impbid1
+      wceq syl5bb df-dc orcom simpll simpr sseldd velsn sylib eqeltrrd pm2.65da
+      simplr eq0rdv ex noel mtbiri wex elex2 sssnm syl biimpcd 0ex snid orbi12d
+      wb mpbiri pm5.74i albii bitri ) DAEZFGZHZFVHIZJZKZALVJVHFSZVHVISZMZKZALAN
+      VMVQAVJVLVPVLVKVKOZMZVJVPVKUAVSVRVKMVJVPVKVRUBVJVRVNVKVOVJVRVNVJVRVNVJVRP
+      ZBVHVTBEZVHIZVKVTWBPZWAFVHWCWAVIIWAFSWCVHVIWAVJVRWBUCVTWBUDZUEBFUFUGWDUHV
+      JVRWBUJUIUKULVNVKFFIFUMVHFFQUNRVJVKVOVKVJVOVKCEVHICUOVJVOVCCFVHUPCVHFUQUR
+      USVOVKFVIIFUTVAVHVIFQVDRVBTTVEVFVG $.
+  $}
+
+  ${
+    $( Excluded middle is equivalent to decidability of ` (/) ` being an
+       element of an arbitrary set.  (Contributed by Jim Kingdon,
+       18-Jun-2022.) $)
+    exmid0el $p |- ( EXMID <-> A. x DECID (/) e. x ) $=
+      ( wem c0 cv wcel wdc wal exmidexmid alrimiv csn wss alimi df-exmid sylibr
+      wi ax-1 impbii ) BCADZEZFZAGZBTASHIUARCJKZTOZAGBTUCATUBPLAMNQ $.
+  $}
+
+  ${
+    $d x y $.
+    $( Excluded middle is equivalent to decidability of membership for two
+       arbitrary sets.  (Contributed by Jim Kingdon, 18-Jun-2022.) $)
+    exmidel $p |- ( EXMID <-> A. x A. y DECID x e. y ) $=
+      ( wem cv wcel wdc wal exmidexmid alrimivv c0 wceq eleq1 dcbid albidv spcv
+      0ex exmid0el sylibr impbii ) CADZBDZEZFZBGZAGZCUCABUBHIUEJUAEZFZBGZCUDUHA
+      JPTJKZUCUGBUIUBUFTJUALMNOBQRS $.
+  $}
+
+  ${
+    $d x y z $.
+    $( Union of complementary parts producing the whole and excluded middle.
```

Comment:
I would write the first sentence as:
> Excluded middle is equivalent to every subset having a complement (i.e., the union of a subset and its relative complement is the whole set).

Reply:
I reworded this as two sentences rather than a sentence with a parenthetical, because I thought there was a tool which grabbed the first sentence. But now I'm not so sure. WRITE THEOREM_LIST seems to take the whole description, so maybe I am imagining it.

I guess I'll leave it as two sentences (hope it doesn't seem confusing that way).

Reply:
Thanks. Just for the record and to clarify why I insisted on the phrase "every subset has a complement": this is a recurring theme: "every subspace of a fin. dim. vector space has a supplement", "every closed subspace of a Hilbert space has a (closed) supplement", the Zassenhaus theorem for "group complements", etc.

Reply:
I'm not familiar with those examples but I'll take your word for it.